File: rust/src/p0010.rs
/*
Project Euler Problem 10

This one was also relatively easy, especially given the work I have
done on my prime number infrastructure.

Problem:

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
*/
use crate::primes::primes;

pub fn p0010() -> i128 {
    return primes::<u64>().take_while(|&p| p < 2000000).sum();
}

File: javascript/src/p0010.js
/**
 * Project Euler Problem 10
 *
 * Yet again, having a good prime number infrastructure comes in handy
 *
 * Problem:
 *
 * The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
 *
 * Find the sum of all the primes below two million.
 *
 * @return {number}
 */
exports.p0010 = function() {
    let answer = 0;
    for (const p of primes.primes(2000000)) {
        answer += p;
    }
    return answer;
};

const primes = require('./lib/primes.js');
